test/e2e/prometheus_test.go

```diff
@@ -318,3 +318,61 @@ func remoteWriteCheckMetrics(ctx context.Context, t *testing.T) {
 		)
 	}
 }
+
+func TestBodySizeLimit(t *testing.T) {
+	const (
+		pvcClaimName       = "prometheus-data"
+		storage            = "20Gi"
+		bodySizeLimitSmall = "1MB"
+	)
+
+	cm := f.MustGetConfigMap(t, clusterMonitorConfigMapName, f.Ns)
+	cmBackup := cm.DeepCopy()
+	cmBackup.ObjectMeta.ResourceVersion = ""
+	cmBackup.ObjectMeta.UID = ""
+	cmBackup.ObjectMeta.CreationTimestamp = metav1.Time{}
+
+	restoreConfig := func() {
+		f.MustCreateOrUpdateConfigMap(t, cmBackup)
+	}
+
+	defer restoreConfig()
+
+	f.PrometheusK8sClient.WaitForQueryReturnFloat(
+		t, 5*time.Minute, `sum(increase(prometheus_target_scrapes_exceeded_body_size_limit_total{job="prometheus-k8s"}[5m]))`,
+		func(v float64) error {
+			if v > 0.0 {
+				return fmt.Errorf("expected prometheus_target_scrapes_exceeded_body_size_limit_total does not increase up but got %v increase in last 5 minutes", v)
+			}
+
+			return nil
+		},
+	)
+
+	data := fmt.Sprintf(`prometheusK8s:
+  logLevel: debug
+  enforcedBodySizeLimit: %s
+  volumeClaimTemplate:
+    metadata:
+      name: %s
+    annotations:
+      openshift.io/cluster-monitoring-drop-pvc: "yes"
+    spec:
+      resources:
+        requests:
+          storage: %s
```

Comment:
I don't think we need persistent storage for the test?

Reply:
I'm going to remove the persistent storage for a test. if it goes well, I will remove that definitely.